tdmpc2/tdmpc2.py

```diff
@@ -23,16 +23,32 @@ def __init__(self, cfg):
 			{'params': self.model._dynamics.parameters()},
 			{'params': self.model._reward.parameters()},
 			{'params': self.model._Qs.parameters()},
-			{'params': self.model._task_emb.parameters() if self.cfg.multitask else []}
-		], lr=self.cfg.lr)
-		self.pi_optim = torch.optim.Adam(self.model._pi.parameters(), lr=self.cfg.lr, eps=1e-5)
+			{'params': self.model._task_emb.parameters() if self.cfg.multitask else []
+			 }
+		], lr=self.cfg.lr, capturable=True)
+		self.pi_optim = torch.optim.Adam(self.model._pi.parameters(), lr=self.cfg.lr, eps=1e-5, capturable=True)
 		self.model.eval()
 		self.scale = RunningScale(cfg)
 		self.cfg.iterations += 2*int(cfg.action_dim >= 20) # Heuristic for large action spaces
 		self.discount = torch.tensor(
 			[self._get_discount(ep_len) for ep_len in cfg.episode_lengths], device='cuda'
 		) if self.cfg.multitask else self._get_discount(cfg.episode_length)
 
+		self.plan_dict = {
+				(True, True): functools.partial(self.plan, t0=True, eval_mode=True),
+				(False, True): functools.partial(self.plan, t0=False, eval_mode=True),
+				(True, False): functools.partial(self.plan, t0=True, eval_mode=False),
+				(False, False): functools.partial(self.plan, t0=False, eval_mode=False),
+				}
```

Comment:
this isn't very elegant but we need to have a separate cudagraph for each `plan` config (unless a tensor is passed).
We could automate that in CudaGraphModule but that would induce some overhead (check the non-tensor inputs). So I'm doing that manually here 
Not terribly sexy, happy to hear thoughts about this!